src/main/java/mate/academy/spring/service/impl/CustomUserDetailsService.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.spring.service.impl;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String userEmail)
+            throws UsernameNotFoundException {
+        User user = userService.findByEmail(userEmail)
+                .orElseThrow(() -> new UsernameNotFoundException("Can't find user " + userEmail));
+        org.springframework.security.core.userdetails.User.UserBuilder builder =
+                org.springframework.security.core.userdetails.User.withUsername(userEmail);
+        builder.password(user.getPassword());
+        builder.authorities(user.getRoles().stream()
+                .map(r -> r.getRoleName().name())
+                .toArray(String[]::new));
```

Comment:
is authentication works correctly? If I'm not wrong you should add prefix `ROLE_` to each role's name